erpnext/accounts/doctype/sales_invoice/sales_invoice.js

```diff
@@ -618,6 +627,19 @@ frappe.ui.form.on('Sales Invoice', {
 			}
 		}
 
+		// discount account
+		frm.fields_dict['items'].grid.get_field('discount_account').get_query = function(doc) {
+			if (erpnext.is_perpetual_inventory_enabled(doc.company)) {
```

Comment:
Remove this check. This has nothing to do with perpetual inventory

Reply:
Done.